mount/mount.go

```diff
@@ -189,6 +189,42 @@ func (d *Datastore) Put(key ds.Key, value []byte) error {
 	return cds.Put(k, value)
 }
 
+// Sync implements Datastore.Sync
+func (d *Datastore) Sync(prefix ds.Key) error {
+	// Sync all mount points below the prefix
+	// Sync the mount point right at (or above) the prefix
+	// Return ErrNoMount if no mount points are synced
+	atLeastOneSync := false
+	for _, m := range d.mounts {
+		if m.Prefix.Equal(prefix) {
+			if err := m.Datastore.Sync(ds.NewKey("/")); err != nil {
+				return err
+			}
+			return nil
+		}
+		if m.Prefix.IsAncestorOf(prefix) {
+			s := strings.TrimPrefix(prefix.String(), m.Prefix.String())
+			k := ds.NewKey(s)
+			if err := m.Datastore.Sync(k); err != nil {
+				return err
+			}
+			return nil
+		}
+
+		if m.Prefix.IsDescendantOf(prefix) {
+			if err := m.Datastore.Sync(ds.NewKey("/")); err != nil {
+				return err
+			}
+			atLeastOneSync = true
+		}
+	}
+
+	if !atLeastOneSync {
+		return ErrNoMount
+	}
```

Comment:
I'd expect syncing a non-existent key to return success: there's nothing there so there's nothing to persist.

`Put` returns an error because the put fails. `Get` just returns `ErrNotFound`.

Reply:
No problem here.

My thought was that if you tried to access an unmounted part of the hierarchy you might want to get an error. It's reasonable (and easier for testing purposes) to say that we did Sync all puts/deletes under that key and there happen to be none there because that key is unmounted.